File: access-checkout-react-native-sdk/src/validation/CardValidationConfig.ts
import type { CardValidationEventListener } from '../index';

export class CardValidationConfig {
  enablePanFormatting?: boolean;
  acceptedCardBrands?: string[];
  validationListener?: CardValidationEventListener;

  constructor({
    validationListener,
    enablePanFormatting = false,
    acceptedCardBrands = [],
  }: CardValidationConfig) {
    this.enablePanFormatting = enablePanFormatting;
    this.validationListener = validationListener;
    this.acceptedCardBrands = acceptedCardBrands;
  }
}

File: access-checkout-react-native-sdk/android/src/main/java/com/worldpay/access/checkout/reactnative/validation/CardValidationConfig.kt
package com.worldpay.access.checkout.reactnative.validation

class CardValidationConfig(
        val baseUrl: String,
        val panId: String,
        val expiryDateId: String,
        val cvcId: String,
        val enablePanFormatting: Boolean,
        val acceptedCardBrands: Array<String>
)

